File: leetcode/452.minimum-number-of-arrows-to-burst-balloons/ts/findminarrowshots.ts
const findMinArrowShots = (points: number[][]): number => {
    if (points.length === 0) return 0

    points.sort((a, b) => {
        if (a[1] !== b[1]) return a[1] - b[1]
        return a[0] - b[0]
    })

    let res = 1
    let bound = points[0][1]

    for (const p of points) {
        if (p[0] <= bound) continue

        res++
        bound = p[1]
    }

    return res
}

File: leetcode/452.minimum-number-of-arrows-to-burst-balloons/cpp/findminarrowshots.cpp
#include <vector>

class Solution
{
public:
    int findMinArrowShots(std::vector<std::vector<int>> &points)
    {
        if (points.size() == 0)
            return 0;

        std::sort(points.begin(), points.end(),
                  [](std::vector<int> a, std::vector<int> b)
                  {
                      if (a[1] != b[1])
                          return a[1] < b[1];
                      return a[0] < b[0];
                  });

        int res = 1;
        int bound = points[0][1];

        for (auto p : points)
        {
            if (p[0] <= bound)
                continue;
            res++;
            bound = p[1];
        }

        return res;
    }
};
